### PR TITLE
fix: useSubmit

### PR DESCRIPTION
**Pull request by yanzhihong23**
## Description

Closes [MF-719](https://mask.atlassian.net/browse/MF-719)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

**Comment by yanzhihong23** [comment]
Added in https://github.com/DimensionDev/Maskbook/pull/6187/commits/e79aeb7e044e852d139be2a5db70912c7edb2a94

**Review by Jack-Works — APPROVED** [review]

**Review by guanbinrui — APPROVED** [review]

File: packages/mask/src/components/CompositionDialog/useSubmit.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ export function useSubmit(onClose: () => void, reason: 'timeline' | 'popup' | 'r
     return useCallback(
         async (info: SubmitComposition) => {
             const { content, encode, target } = info
-            const currentProfile = globalUIState.profiles.value?.[0].identifier
+            const currentProfile = globalUIState.profiles.value?.[0]?.identifier
```
Comment (guanbinrui) [inline packages/mask/src/components/CompositionDialog/useSubmit.ts:20]: Why is the type checker not working here? @yanzhihong23

Comment (Jack-Works) [reply packages/mask/src/components/CompositionDialog/useSubmit.ts:20]: Because TypeScript by default doesn't treat the Array index as undefined.

Comment (Jack-Works) [reply packages/mask/src/components/CompositionDialog/useSubmit.ts:20]: You need either Array.prototype.at or enable --noUncheckedIndexedAccess flag

image

**Pull request merged**